controller.js

```diff
@@ -254,6 +254,67 @@ module.exports.createDish = (req, res) => {
     });
 };
 
+module.exports.bulkCreateDish = async (req, res) => {
+  let ids = req.body.ids;
+
+  const menuData = {
+    name: req.body.name,
+    restaurantId: req.user.restaurantId,
+    published: true,
+  };
+
+  const menu = await Menu.create(menuData).then((menu) => {
+    //wrap this into promise
+    var promises = ids.map(async (id) => {
+      Dish.findByPk(id, {
+        include: [
+          { model: Category, attributes: ["name"] },
+          { model: Tag, as: "Tags", attributes: ["id"] },
+        ]
+      }).then((originalDish) => {
+        getOrCreateCategory(originalDish.Category.name, menu.id).then((categoryId) => {
+          const dishData = {
+            name: originalDish.name,
+            description: originalDish.description,
+            addons: originalDish.addons,
+            canRemove: originalDish.canRemove,
+            notes: originalDish.notes,
+            tableTalkPoints: originalDish.tableTalkPoints,
+            restaurantId: originalDish.restaurantId,
+            categoryId: categoryId,
+            menuId: menu.id,
+            price: originalDish.price,
+          };
+
+          tagIds = [];
+
+          originalDish.Tags.forEach((tag) => {
+            tagIds.push(tag.id);
+          });
+
+          Dish.create(dishData)
+            .then((dish) => {
+              dish.setTags(tagIds).then(() => {});
+            })
+            .catch((err) => {
+              console.error(err);
+              res.status(500).send({
+                message: err.message || "Dish could not be created",
+              });
+            });
+        });
+      });
+
+      return new Promise((res, rej) => {res({})});
+    });
+
+    Promise.all(promises).then((results) => {
+      console.log(promises);
+      res.send(menu);
+    });
+  });
+};
+
```

Comment:
We just need to make sure to delete the menu if an error occurs during creation

Reply:
fixed

Reply:
Shall we use a Sequelize transaction for bulk stuff like this?

Reply:
That's true - might make more sense to do that.  I can look into this and update accordingly.